app/src/main/java/swati4star/createpdf/fragment/TextToPdfFragment.java

```diff
@@ -477,7 +475,7 @@ public void onRequestPermissionsResult(int requestCode,
             case PERMISSION_REQUEST_WRITE_EXTERNAL_STORAGE_RESULT: {
                 if (grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                     mPermissionGranted = true;
-                    openCreateTextPdf();
+                    selectTextFile();
```

Comment:
is it also changed by mistake?

Reply:
No, I though about something.
let me think about it :thinking:

Reply:
when user clicks on select images button (selectTextFile() method)
this is the method where we check if user has permission or not.
if he has, then everything will work as normal
if he doesn't have permission, then we will request permission and return from this method and now onRequestPermissionsResult() will be called, if user grants the permission then selectTextFile() method is where we want to get back as earlier we returned from this method immediately after requesting permission.
I hope it's clear

Reply:
Sure. Makes sense :)